src/subapps/projects/containers/ActivityResourcesContainer.tsx

```diff
@@ -1,52 +1,150 @@
 import * as React from 'react';
 import { useNexusContext } from '@bbp/react-nexus';
-import { Resource, ResourceLink } from '@bbp/nexus-sdk';
+import { Resource, DEFAULT_ELASTIC_SEARCH_VIEW_ID } from '@bbp/nexus-sdk';
 
 import { displayError } from '../components/Notifications';
 import ResourcesPane from '../components/ResourcesPane';
 import ResourcesList from '../components/ResourcesList';
-import { isActivityResourceLink } from '../utils';
 import fusionConfig from '../config';
 import { CodeResourceData } from '../components/LinkCodeForm';
+import ResourcesSearch from '../components/ResourcesSearch';
+import { ActivityResource } from '../views/ActivityView';
 
 const ActivityResourcesContainer: React.FC<{
   orgLabel: string;
   projectLabel: string;
-  activityId: string;
+  activity: ActivityResource;
   linkCodeToActivity: (codeResourceId: string) => void;
-}> = ({ orgLabel, projectLabel, activityId, linkCodeToActivity }) => {
+}> = ({ orgLabel, projectLabel, activity, linkCodeToActivity }) => {
   const nexus = useNexusContext();
+
   const [resources, setResources] = React.useState<Resource[]>([]);
+  const [search, setSearch] = React.useState<string>();
+  const [typeFilter, setTypeFilter] = React.useState<string[]>();
+  const [busy, setBusy] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    fetchLinkedResources();
```

Comment:
can we move this and fetchLinkedResources method in to custom hook in a separate file?  It will helps us to keep this container short and clean.